internal/collector/pod.go

```diff
@@ -32,10 +33,35 @@ import (
 
 const nodeUnreachablePodReason = "NodeLost"
 
+type containerTerminationInfo struct {
+	podContainerReasonCounter sync.Map
+}
+
+func (c *containerTerminationInfo) add(pod, container, reason string, isReason bool) float64 {
+	obj, _ := c.podContainerReasonCounter.LoadOrStore(pod, &sync.Map{})
+	containerReasonCounter := obj.(*sync.Map)
+	obj, _ = containerReasonCounter.LoadOrStore(container, &sync.Map{})
+	reasonCounter := obj.(*sync.Map)
+	obj, _ = reasonCounter.LoadOrStore(reason, 0.0)
+	counter := obj.(float64)
+	if isReason {
+		reasonCounter.Store(reason, counter+1.0)
+		return counter + 1.0
+	}
+	return counter
+}
+
+func (c *containerTerminationInfo) cleanup(p *v1.Pod) {
+	if p.DeletionTimestamp != nil {
+		c.podContainerReasonCounter.Delete(p.Name)
+	}
+}
+
 var (
-	descPodLabelsDefaultLabels = []string{"namespace", "pod"}
-	containerWaitingReasons    = []string{"ContainerCreating", "CrashLoopBackOff", "CreateContainerConfigError", "ErrImagePull", "ImagePullBackOff"}
-	containerTerminatedReasons = []string{"OOMKilled", "Completed", "Error", "ContainerCannotRun"}
+	descPodLabelsDefaultLabels    = []string{"namespace", "pod"}
+	containerWaitingReasons       = []string{"ContainerCreating", "CrashLoopBackOff", "CreateContainerConfigError", "ErrImagePull", "ImagePullBackOff"}
+	containerTerminatedReasons    = []string{"OOMKilled", "Completed", "Error", "ContainerCannotRun"}
```

Comment:
These terminated reasons which is only for runtime type [`docker`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockershim/docker_container.go#L352), Do we need to mark it?